CONTRIBUTING.md

```diff
@@ -173,6 +173,47 @@ The `legacy_foo` arg was removed
 :::
 ```
 
+## Style and idioms
+
+For the most part, we just accept whatever the code formatters do, so there
+isn't much style to enforce.
+
+Some miscellanous style, idioms, and conventions we have are:
+
+### Markdown/Sphinx Style
+
+* Use colons for prose sections of text, e.g. `:::{note}`, not backticks.
+* Use backticks for code blocks.
+* Max line length: 100.
+
+### BUILD/bzl Style
+
+* When a macro generates public targets, use a dot (`.`) to separate the
+  user-provided name from the generted name. e.g. `foo(name="x")` generates
+  `x.test`. The `.` is our convention to communicate that it's a generated
+  target, and thus one should look for `name="x"` when searching for the
+  definition.
+* The different build phases and interface/implementation pieces of rules should
+  be in different files. Stated another way: loading code for thing shouldn't
+  incur loading code that isn't relevant.
+  * Providers should be in their own files. This allows implementing a custom
+    rule that implements the provider without loading an existing
+    implementation.
+  * The repository phase shouldn't `load()` code that is only relevant to the
+    loading-phase or subsequent phases. Vice-versa, loading-phase code should
+    not load any repository-phase code.
+  * Generally, one rule per file. The goal is that defining an e.g. library
+    shouldn't incur loading all the code for binaries, tests, packaging, etc.
+* The public access point for anything should be a separate file that re-exports
+  the specific symbols that are public. This ensures our public API is well
+  defined and prevents accidentally exposing a package-private symbol as a
+  public symbol.
```

Comment:
The documentation about the symbol should be put in there - i.e. the docstrings of the file next to the implementation do not do anything and we should add proper user facing documentation next to the exports.

Reply:
Rephrased to incorporate that

Reply:
done